spec/services/product_tag_rules_filterer_spec.rb

```diff
@@ -50,9 +50,98 @@
     }
     let(:filterer) { described_class.new(distributor, customer, variants_relation) }
 
-    context "when the distributor has no rules" do
-      it "returns the relation unchanged" do
-        expect(filterer.call).to eq variants_relation
+    describe "#call" do
+      before do
+        v1
+        v2
+        v3
+        v4
+      end
+
+      context "when the distributor has no rules" do
+        it "returns the relation unchanged" do
+          expect(filterer.call).to eq variants_relation
+        end
+      end
+
+      context "with hide rule" do
+        it "hides the variant matching the rule" do
+          customer.update_attribute(:tag_list, hide_rule.preferred_customer_tags)
+          variant_hidden_by_rule.update_attribute(:tag_list, hide_rule.preferred_variant_tags)
+
+          expect(filterer.call).not_to include(variant_hidden_by_rule.variant)
+        end
+
+        context "with mutiple conflicting rules" do
+          it "applies the hide rule" do
+            # Customer has show rule tag and hide rule tag
+            customer.update_attribute(:tag_list,
+                                      [hide_rule.preferred_customer_tags,
+                                       show_rule.preferred_customer_tags])
+            # Variant has show rule tag and hide rule tag
+            variant_hidden_by_rule.update_attribute(:tag_list,
+                                                    [hide_rule.preferred_variant_tags,
+                                                     hide_rule.preferred_variant_tags,])
```

Comment:
> Variant has show rule tag and hide rule tag

Should that be:
```suggestion
                                                     show_rule.preferred_variant_tags])
```

Reply:
Nice catch ! turns out the test was actually wrong :man_facepalming: